[sdl2-mixer] with feature libmodplug build failure

### Operating system

Windows 11 10.0.22631.3085 x64 Enterprise

### Compiler

Visual Studio Community 2022 Version: [17.8.6]  VS Tools Version: [14.37.32822] 
Microsoft (R) C/C++ Optimizing Compiler Version [19.37.32826.1] for x64 

Vcpkg Version [2024-02-07-8a83681f921b10d86ae626fd833c253f4f8c355b] 
Vcpkg Scripts Version [2024-02-07-783c0fe1dcdf4616930f336377d65028ffb60f3e] 

### Steps to reproduce the behavior

```Shell
`.\vcpkg.exe install sdl2-mixer[libmodplug]`
```

### Failure logs

Computing installation plan...
The following packages will be built and installed:
    sdl2-mixer[core,libmodplug]:x64-windows@2.8.0
Installing 1/1 sdl2-mixer[core,libmodplug]:x64-windows@2.8.0...
Building sdl2-mixer[core,libmodplug]:x64-windows@2.8.0...
-- [OVERLAY] Loading triplet configuration from: D:\vcpkg-overlays\triplets\x64-windows.cmake
-- Using cached libsdl-org-SDL_mixer-release-2.8.0.tar.gz.
-- Extracting source D:/vcpkg/downloads/libsdl-org-SDL_mixer-release-2.8.0.tar.gz
-- Applying patch fix-pkg-prefix.patch
-- Using source at D:/vcpkg/buildtrees/sdl2-mixer/src/ease-2.8.0-5d1a2b7a04.clean
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: D:/vcpkg/buildtrees/sdl2-mixer/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\vcpkg\buildtrees\sdl2-mixer\config-x64-windows-dbg-CMakeCache.txt.log
      D:\vcpkg\buildtrees\sdl2-mixer\config-x64-windows-rel-CMakeCache.txt.log
      D:\vcpkg\buildtrees\sdl2-mixer\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/sdl2-mixer/portfile.cmake:31 (vcpkg_cmake_configure)
  scripts/ports.cmake:170 (include)


error: building sdl2-mixer:x64-windows failed with: BUILD_FAILED
Elapsed time to handle sdl2-mixer:x64-windows: 2.1 s
Total install time: 2.1 s
RESULTS
  sdl2-mixer:x64-windows: BUILD_FAILED: 2.1 s

SUMMARY FOR x64-windows
  BUILD_FAILED: 1

[config-x64-windows-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/14215246/config-x64-windows-dbg-CMakeCache.txt.log)
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/14215247/config-x64-windows-out.log)
[config-x64-windows-rel-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/14215248/config-x64-windows-rel-CMakeCache.txt.log)

### Additional context

Found in file "config-x64-windows-out.log":

>CMake Error at CMakeLists.txt:725 (target_link_libraries):
>  Target "SDL2_mixer" links to:
>
>    libxmp::libxmp
>
>  but the target was not found.  Possible reasons include:
>
>    * There is a typo in the target name.
>    * A find_package call is missing for an IMPORTED target.
>    * An ALIAS target is missing.



I verified I do have libxmp installed:
>`.\vcpkg.exe list libxmp`
>libxmp:x64-windows                                4.6.0#1             Libxmp is a library that renders module files to...
>libxmp[depackers]:x64-windows                                         Enable archive depackers.
>libxmp[prowizard]:x64-windows                                         Enable ProWizard format loaders.

A depend-info does not show libxmp is required but then wavpack is???
>`.\vcpkg.exe depend-info sdl2-mixer[libmodplug]`
>vcpkg-cmake:
>vcpkg-cmake-config:
>libogg: vcpkg-cmake, vcpkg-cmake-config
>libmodplug: vcpkg-cmake
>libvorbis: libogg, vcpkg-cmake, vcpkg-cmake-config
>sdl2: vcpkg-cmake, vcpkg-cmake-config
>wavpack: vcpkg-cmake, vcpkg-cmake-config
>sdl2-mixer[libmodplug]: libmodplug, libvorbis, sdl2, vcpkg-cmake, vcpkg-cmake-config, wavpack

I am using an overlay triplet made from the default x64-windows that specifies a specific version of the MSVC Compiler as follows:
>set(VCPKG_TARGET_ARCHITECTURE x64)
>set(VCPKG_CRT_LINKAGE dynamic)
>set(VCPKG_LIBRARY_LINKAGE dynamic)
>set(VCPKG_PLATFORM_TOOLSET_VERSION "14.37.32822")

I did try using the default x64-windows triplet with the same results. The overlay is for the work-around for jasper port - see https://github.com/microsoft/vcpkg/issues/36103#issuecomment-1929190155

--- a/ports/sdl2-mixer/portfile.cmake
+++ b/ports/sdl2-mixer/portfile.cmake
@@ -28,6 +28,8 @@ if("fluidsynth" IN_LIST FEATURES)
     list(APPEND EXTRA_OPTIONS "-DPKG_CONFIG_EXECUTABLE=${PKGCONFIG}")
 endif()
 
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" BUILD_SHARED)
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
@@ -43,6 +45,7 @@ vcpkg_cmake_configure(
         -DSDL2MIXER_MIDI_NATIVE=OFF
         -DSDL2MIXER_MIDI_TIMIDITY=OFF
         -DSDL2MIXER_MP3_DRMP3=OFF
+        -DSDL2MIXER_MOD_XMP_SHARED=${BUILD_SHARED}
 )
 
 vcpkg_cmake_install()

--- a/ports/sdl2-mixer/vcpkg.json
+++ b/ports/sdl2-mixer/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "sdl2-mixer",
   "version": "2.8.0",
+  "port-version": 1,
   "description": "Multi-channel audio mixer library for SDL.",
   "homepage": "https://github.com/libsdl-org/SDL_mixer",
   "license": "Zlib",
@@ -36,7 +37,8 @@
     "libmodplug": {
       "description": "Use libmodplug to play MOD audio format.",
       "dependencies": [
-        "libmodplug"
+        "libmodplug",
+        "libxmp"
       ]
     },
     "mpg123": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7834,7 +7834,7 @@
     },
     "sdl2-mixer": {
       "baseline": "2.8.0",
-      "port-version": 0
+      "port-version": 1
     },
     "sdl2-mixer-ext": {
       "baseline": "2.6.0",

--- a/versions/s-/sdl2-mixer.json
+++ b/versions/s-/sdl2-mixer.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "b28cae64adf73bec946de9f037724763eb2ef1b2",
+      "version": "2.8.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "21406093a22cbd3a55befe2ff61f81011d055036",
       "version": "2.8.0",
       "port-version": 0

